includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2214,11 +2214,17 @@ public function get_level3_data_from_order( WC_Order $order ): array {
 				$product_id = substr( sanitize_title( $item->get_name() ), 0, 12 );
 			}
 
-			$description     = substr( $item->get_name(), 0, 26 );
-			$quantity        = ceil( $item->get_quantity() );
-			$unit_cost       = WC_Payments_Utils::prepare_amount( $subtotal / $quantity, $currency );
-			$tax_amount      = WC_Payments_Utils::prepare_amount( $item->get_total_tax(), $currency );
-			$discount_amount = WC_Payments_Utils::prepare_amount( $subtotal - $item->get_total(), $currency );
+			$description = substr( $item->get_name(), 0, 26 );
+			$quantity    = ceil( $item->get_quantity() );
+			$tax_amount  = WC_Payments_Utils::prepare_amount( $item->get_total_tax(), $currency );
+			if ( $subtotal >= 0 ) {
+				$unit_cost       = WC_Payments_Utils::prepare_amount( $subtotal / $quantity, $currency );
+				$discount_amount = WC_Payments_Utils::prepare_amount( $subtotal - $item->get_total(), $currency );
```

Comment:
I wonder if it is possible for `$subtotal` to be less than `$item->get_total()`? Would it require to cover such case as a separate flow?

Reply:
It's the order item total, not the complete order total: it's possible to create products with negative prices.

Reply:
I'll rephrase my question: is it possible that `$discount_amount` calculated here will be negative? I.e.are there tests to ensure that item's subtotal is less than total? 🙂

Reply:
Thanks @dmvrtx and @kalessil I'll look into this

Reply:
> I'll rephrase my question: is it possible that `$discount_amount` calculated here will be negative? I.e.are there tests to ensure that item's subtotal is less than total? 🙂

Looking into `WC_Order_Item_Product` class, I found that the `set_total` method includes some logic that prevents the item's `subtotal` to be less than `total` (full code [here](https://woocommerce.github.io/code-reference/files/woocommerce-includes-class-wc-order-item-product.html#source-view.118) for reference):
````
// Subtotal cannot be less than total.
if ( '' === $this->get_subtotal() || $this->get_subtotal() < $this->get_total() ) {
	    $this->set_subtotal( $value );
}
````
If I understand that logic correctly, `$subtotal` cannot be less than `$item->get_total()` and $discount_amount will be 0 or more than 0.